SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -104,44 +87,25 @@ bool ComponentFactory::FillSchemaObject(Schema_Object* ComponentObject, UObject*
 
 bool ComponentFactory::FillHandoverSchemaObject(Schema_Object* ComponentObject, UObject* Object, const FClassInfo& Info, const FHandoverChangeState& Changes, bool bIsInitialData, TArray<Schema_FieldId>* ClearedIds /* = nullptr */)
 {
-	bool bWroteSomething = false;
-
 	for (uint16 ChangedHandle : Changes)
 	{
 		check(ChangedHandle > 0 && ChangedHandle - 1 < Info.HandoverProperties.Num());
 		const FHandoverPropertyInfo& PropertyInfo = Info.HandoverProperties[ChangedHandle - 1];
 
 		const uint8* Data = (uint8*)Object + PropertyInfo.Offset;
-		FUnresolvedObjectsSet UnresolvedObjects;
-
-		AddProperty(ComponentObject, ChangedHandle, PropertyInfo.Property, Data, UnresolvedObjects, ClearedIds);
-
-		if (UnresolvedObjects.Num() == 0)
-		{
-			bWroteSomething = true;
-		}
-		else
-		{
-			if (!bIsInitialData)
-			{
-				// Don't send updates for fields with unresolved objects, unless it's the initial data,
-				// in which case all fields should be populated.
-				Schema_ClearField(ComponentObject, ChangedHandle);
-			}
 
-			PendingHandoverUnresolvedObjectsMap.Add(ChangedHandle, UnresolvedObjects);
-		}
+		AddProperty(ComponentObject, ChangedHandle, PropertyInfo.Property, Data, ClearedIds);
 	}
 
-	return bWroteSomething;
+	return Changes.Num() > 0;
```

Comment:
Nit: Why not keep bWroteSomething = true; inside the loop and return it?

Reply:
Yeah, I guess that's more self-explanatory.